crates/flow-web/src/lib.rs

```diff
@@ -148,3 +154,60 @@ pub fn extend_read_bundle(input: JsValue) -> Result<JsValue, JsValue> {
 
     serde_wasm_bindgen::to_value(&output).map_err(|err| JsValue::from_str(&format!("{err:?}")))
 }
+
+#[wasm_bindgen]
+pub fn get_resource_config_pointers(input: JsValue) -> Result<JsValue, JsValue> {
+    #[derive(serde::Deserialize)]
+    #[serde(rename_all = "camelCase", deny_unknown_fields)]
+    struct Input {
+        spec: String,
+    }
+    let Input { spec } = serde_wasm_bindgen::from_value(input)
+        .map_err(|err| JsValue::from_str(&format!("Invalid JSON: {:?}", err)))?;
+
+    let pointers = tables::utils::pointer_for_schema(&spec.as_ref())
+        .map_err(|err| JsValue::from_str(&format!("Failed getting pointers: {:?}", err)))?;
+
+    #[derive(Serialize, Deserialize)]
+    struct Output {
+        pointers: ResourceSpecPointers,
+    }
+    serde_wasm_bindgen::to_value(&Output { pointers })
+        .map_err(|err| JsValue::from_str(&format!("{err:?}")))
+}
+
+#[wasm_bindgen]
+pub fn update_materialization_resource_spec(input: JsValue) -> Result<JsValue, JsValue> {
+    let input: serde_json::Value = ::serde_wasm_bindgen::from_value(input)
+        .map_err(|err| JsValue::from_str(&format!("invalid JSON: {:?}", err)))?;
+
+    #[derive(serde::Deserialize)]
+    #[serde(rename_all = "camelCase", deny_unknown_fields)]
+    struct Input {
+        source_capture: SourceCapture,
+        resource_spec: serde_json::Value,
+        resource_spec_pointers: ResourceSpecPointers,
+        collection_name: String,
+    }
+
+    let Input {
+        source_capture,
+        resource_spec,
+        resource_spec_pointers,
+        collection_name,
+    } = serde_json::from_value(input)
+        .map_err(|err| JsValue::from_str(&format!("invalid input: {:?}", err)))?;
+
+    let mut resource_spec = resource_spec.clone();
+    tables::utils::update_materialization_resource_spec(
+        &source_capture,
+        &mut resource_spec,
+        &resource_spec_pointers,
+        collection_name.as_ref(),
+    )
+    .map_err(|err| JsValue::from_str(&format!("Failed updating resource spec: {:?}", err)))?;
+
+    // Outputting as a string as I just could NOT get it to return JSON correctly
+    let output = serde_json::to_string(&resource_spec).unwrap();
+    serde_wasm_bindgen::to_value(&{ output }).map_err(|err| JsValue::from_str(&format!("{err:?}")))
```

Comment:
I quit trying to get this to return a JSON object to JS and just returned a string and parse it in the JS.

I kept getting the `Value` back (map of key values) and got frustrated and stopped trying.

![wasmResponse](https://github.com/user-attachments/assets/2b9f2d4f-1f02-42a5-8bae-0474311d1c48)